libkbfs/tlf_journal.go

```diff
@@ -402,33 +416,134 @@ func (j *tlfJournal) flush(ctx context.Context) (err error) {
 }
 
 func (j *tlfJournal) flushOneBlockOp(ctx context.Context) (bool, error) {
-	j.lock.Lock()
-	defer j.lock.Unlock()
-	return j.blockJournal.flushOne(ctx, j.delegateBlockServer, j.tlfID)
+	j.flushLock.Lock()
+	defer j.flushLock.Unlock()
+
+	o, e, data, serverHalf, err := func() (
```

Comment:
Ah, yeah, actually we might want to not let panics in the background goroutine kill the entire daemon (similar to how fuse operations are handled). So added a TODO for that. But those anonymous functions are pretty ugly, so I just moved those into functions on `*tlfJournal`.